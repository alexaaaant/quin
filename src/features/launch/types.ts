export interface ILaunch {
  id: string;
  name: string;
  latitude: string;
  longitude: string;
}

export interface ILaunchOriginal {
  id: string;
  name: string;
  pad: {
    latitude: string;
    longitude: string;
  };
}
