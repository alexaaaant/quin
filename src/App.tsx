const App: React.FC = () => {
  return <div className="App">New application</div>;
};

export default App;
