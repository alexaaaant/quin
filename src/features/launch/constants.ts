export const MAIN_API_URL = 'https://lldev.thespacedevs.com/2.2.0';
export const LAUNCH = '/launch';

export const ACTIONS = {
  LAUNCH_ALL: 'launch/all'
};
