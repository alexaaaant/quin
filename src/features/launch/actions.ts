import { createAsyncThunk } from '@reduxjs/toolkit';
import { launchesApi } from '../../services/api/launches';
import { ACTIONS } from './constants';
import { IGetAllLaunchesParams } from './types';

export const fetchAllLaunches = createAsyncThunk(
  ACTIONS.LAUNCH_ALL,
  async (params: IGetAllLaunchesParams) => {
    const response = await launchesApi.getAllLaunches(params);
    return response;
  }
);
