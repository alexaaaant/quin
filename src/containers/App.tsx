import { useEffect } from 'react';
import { useDispatch } from '../common/hooks/useDispatch';
import { fetchAllLaunches } from '../features/launch/actions';
import { MapWrapper } from './MapWrapper';

const App: React.FC = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchAllLaunches())
      .then(() => console.log())
      .catch(() => console.log());
  }, []);

  return (
    <div className="App">
      <MapWrapper />
    </div>
  );
};

export default App;
