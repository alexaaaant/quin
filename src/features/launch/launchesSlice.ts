import { createSlice } from '@reduxjs/toolkit';
import { ILaunch } from './types';
import { fetchAllLaunches } from './actions';

export interface LaunchesState {
  launches: Record<ILaunch['id'], ILaunch>;
}

const initialState: LaunchesState = {
  launches: {}
};

const launchesSlice = createSlice({
  name: 'launch',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(fetchAllLaunches.fulfilled, (state, action) => {
      action.payload.results.forEach((launch) => {
        console.log(launch);
        state.launches[launch.id] = {
          id: launch.id,
          name: launch.name,
          latitude: launch.pad.latitude,
          longitude: launch.pad.longitude
        };
      });
    });
  }
});

// export const {} = launchesSlice.actions;

export default launchesSlice.reducer;
