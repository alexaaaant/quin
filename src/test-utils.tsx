import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import launchesSlice from './features/launch/launchesSlice';

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => {
  const store = configureStore({
    reducer: {
      launches: launchesSlice
    }
  });

  return <Provider store={store}>{children}</Provider>;
};

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const scheduler = typeof setImmediate === 'function' ? setImmediate : setTimeout;

async function flushPromises(): Promise<void> {
  return await new Promise((resolve) => scheduler(resolve));
}

// eslint-disable-next-line import/export
export * from '@testing-library/react';
// eslint-disable-next-line import/export
export { customRender as render, flushPromises };
