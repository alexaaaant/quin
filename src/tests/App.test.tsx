import App from '../containers/App';
import { LAUNCH } from '../features/launch/constants';
import { render } from '../test-utils';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

export const handlers = [
  rest.get(`*${LAUNCH}*`, (req, res, ctx) => {
    return res(ctx.json([{ id: '1', name: '2', pad: { latitude: 1, longitude: 2 } }]));
  })
];

const server = setupServer(...handlers);

beforeAll(() => server.listen());

afterEach(() => server.resetHandlers());

afterAll(() => server.close());

describe('App', () => {
  it('makes a snapshot', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
