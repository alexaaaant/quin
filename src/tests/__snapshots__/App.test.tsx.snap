// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App makes a snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      style="display: flex; height: 100%;"
    >
      <div>
        spinner
      </div>
    </div>
  </div>
</div>
`;
